shell/pages/c/_cluster/apps/charts/chart.vue

```diff
@@ -115,7 +115,9 @@ export default {
 
   watch: {
     '$route.query'(neu, old) {
-      if ( !isEqual(neu, old) ) {
+      // If the query changes, refetch the chart
+      // When going back to app list, the query is empty and we don't want to refetch
+      if ( !isEqual(neu, old) && Object.keys(neu).length > 0 ) {
```

Comment:
I was wondering if we should update the same logic in `shell/pages/c/_cluster/apps/charts/install.vue` as well?